src/cluster_legacy.c

```diff
@@ -6064,6 +6067,7 @@ unsigned int delKeysInSlot(unsigned int hashslot) {
     }
     kvstoreReleaseDictIterator(kvs_di);
 
+    server.server_del_keys_in_slot = 0;
```

Comment:
The `propagateNow()` is triggered by the call to `postExecutionUnitOperations()` above, but only if there are no nested execution units. If `delKeysInSlot is called from inside another execution unit, this will not work. Maybe we should assert here `assert(server.execution_nesting == 0)`?

Reply:
adding a assert here seem not harm, i can add it later.